pkg/codec/byte_to_boolean_modifier_test.go

```diff
@@ -0,0 +1,206 @@
+		offchain := reflect.New(convertedType).Elem()
+		offchain.FieldByName("A").SetString("nested")
+		nestedStruct := offchain.FieldByName("B")
+		// Set nested field C (off-chain) to bool true.
+		nestedStruct.FieldByName("C").SetBool(true)
+
+		onChainVal, err := converter.TransformToOnChain(offchain.Interface(), "")
+		require.NoError(t, err)
+		expected := outer{A: "nested", B: inner{C: 1}}
+		assert.Equal(t, expected, onChainVal)
+	})
+
+	t.Run("TransformToOffChain converts on-chain uint8 to off-chain bool", func(t *testing.T) {
+		converter := codec.NewByteToBooleanModifier([]string{"B"})
+		_, err := converter.RetypeToOffChain(reflect.TypeOf(testStruct{}), "")
+		require.NoError(t, err)
+
+		// On-chain instance with B=0 (false).
+		onChain := testStruct{A: "hello", B: 0}
+		offChainVal, err := converter.TransformToOffChain(onChain, "")
+		require.NoError(t, err)
+
+		offchain := reflect.ValueOf(offChainVal)
+		assert.Equal(t, "hello", offchain.FieldByName("A").String())
+		// Field B should now be false.
+		assert.False(t, offchain.FieldByName("B").Bool())
+	})
+
+	t.Run("TransformToOffChain converts nested field uint8 to off-chain bool", func(t *testing.T) {
+		type inner struct {
+			F uint8
+		}
+		type outer struct {
+			D string
+			E inner
+		}
+		converter := codec.NewByteToBooleanModifier([]string{"E.F"})
+		_, err := converter.RetypeToOffChain(reflect.TypeOf(outer{}), "")
+		require.NoError(t, err)
+
+		// On-chain instance with nested field B.C set to 1 (true).
+		onChain := outer{D: "off", E: inner{F: 0}}
+		offChainVal, err := converter.TransformToOffChain(onChain, "")
+		require.NoError(t, err)
+		offchain := reflect.ValueOf(offChainVal)
+		assert.Equal(t, "off", offchain.FieldByName("D").String())
+		// Nested field B.C should now be false.
+		nestedStruct := offchain.FieldByName("E")
+		assert.False(t, nestedStruct.FieldByName("F").Bool())
+	})
+
+	t.Run("TransformToOnChain and TransformToOffChain work on pointer fields", func(t *testing.T) {
+		converter := codec.NewByteToBooleanModifier([]string{"B"})
+		convertedType, err := converter.RetypeToOffChain(reflect.TypeOf(ptrTestStruct{}), "")
+		require.NoError(t, err)
+
+		// Create an off-chain instance.
+		offchainPtr := reflect.New(convertedType)
+		offchain := offchainPtr.Elem()
+		offchain.FieldByName("A").SetString("pointer")
+		boolVal := true
+		offchain.FieldByName("B").Set(reflect.ValueOf(&boolVal))
+
+		onChainVal, err := converter.TransformToOnChain(offchainPtr.Elem().Interface(), "")
+		require.NoError(t, err)
+
+		expected := ptrTestStruct{
+			A: "pointer",
+			B: ptrUint8(1),
+		}
+		assert.True(t, deepEqualPtrTestStruct(onChainVal, expected))
+
+		offChainVal2, err := converter.TransformToOffChain(expected, "")
+		require.NoError(t, err)
+		offchain2 := reflect.ValueOf(offChainVal2)
+		assert.Equal(t, "pointer", offchain2.FieldByName("A").String())
+
+		b, ok := offchain2.FieldByName("B").Interface().(*bool)
+		require.True(t, ok)
+		assert.True(t, *b)
+	})
```

Comment:
Just a note to add a test for slices if we do want to support that functionality. Otherwise, you can ignore this